src/v2/BaseLoginRouter.js

```diff
@@ -207,25 +222,32 @@ export default Router.extend({
         });
     }
 
-    // introspect stateToken when widget is bootstrap with state token
-    // and remove it from `settings` afterwards as IDX response always has
-    // state token (which will be set into AppState)
+    // Remove stateToken and other attributes to sanitize settings object.
+    // IDX response always has stateToken, that will be set into AppState)
+    const renderCallback = (idxResp) => {
+      this.unsetAttributes();
```

Comment:
We may also want to check the existence and set `widget-state-token` here, as the custom hosted interaction code flow starts without stateToken. The first available one will be here (when idx response resolves).

Reply:
nice catch 👍